flow/layers/backdrop_filter_layer.cc

```diff
@@ -54,31 +54,9 @@ void BackdropFilterLayer::Paint(PaintContext& context) const {
   TRACE_EVENT0("flutter", "BackdropFilterLayer::Paint");
   FML_DCHECK(needs_painting(context));
 
-  AutoCachePaint save_paint(context);
-  save_paint.setBlendMode(blend_mode_);
-  if (context.leaf_nodes_builder) {
-    // Note that we perform a saveLayer directly on the
-    // leaf_nodes_builder here similar to how the SkCanvas
-    // path specifies the kLeafNodesCanvas below.
-    // See https:://flutter.dev/go/backdrop-filter-with-overlay-canvas
-    context.leaf_nodes_builder->saveLayer(&paint_bounds(),
-                                          save_paint.dl_paint(), filter_.get());
-
-    PaintChildren(context);
-
-    context.leaf_nodes_builder->restore();
-  } else {
-    auto sk_filter = filter_ ? filter_->skia_object() : nullptr;
-    Layer::AutoSaveLayer save = Layer::AutoSaveLayer::Create(
-        context,
-        SkCanvas::SaveLayerRec{&paint_bounds(), save_paint.sk_paint(),
-                               sk_filter.get(), 0},
-        // BackdropFilter should only happen on the leaf nodes canvas.
-        // See https:://flutter.dev/go/backdrop-filter-with-overlay-canvas
-        AutoSaveLayer::SaveMode::kLeafNodesCanvas);
-
-    PaintChildren(context);
-  }
+  auto save = context.state_stack.saveWithBackdropFilter(&paint_bounds(),
```

Comment:
If the parent node is OpacityLayer, here we need set the alpha of paint.

Reply:
the test case is `TEST_F(BackdropFilterLayerTest, OpacityInheritance)`

Reply:
Opacity inheritance is not really implemented yet - there is a skeleton of a concept for it, but it is not ready for review, just an inkling of where I'm headed. I'm still working out what is the best way to manage it with the new mechanism and what the API for layers will look like.